File: Python/MaximumDepthOfBinaryTree.py
class TreeNode:
    
    def __init__(self,val,left,right) -> None:
        self.val = val
        self.left = left
        self.right = right

def maxDepth(root):

    def findMaxDepth(node):
        if node is None:
            return 0
        
        leftSubTree = findMaxDepth(node.left)
        rightSubTree = findMaxDepth(node.right)

        return 1+ max(leftSubTree,rightSubTree)


    maxDepth = findMaxDepth(root)
    return maxDepth

    

File: Javascript/MaximumDepthOfBinaryTree.js

function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? null : left
    this.right = (right === undefined) ? null : right
}

var maxDepth = function(root){
    if(root === null){
        return 0
    }

    return findMaxDepth(root)

}


function findMaxDepth(node){

    const leftSubTree = maxDepth(node.left)
    const rightSubTree = maxDepth(node.right)

    return 1 + Math.max(leftSubTree, rightSubTree)
}